File: discord_receive.py
import discord
import asyncio
import requests
import emoji

class RhostRelayClient(discord.Client):
    async def on_ready(self):
        print('Logged on as', self.user)

    async def on_message(self, message):
        # don't respond to ourselves
        if message.author == self.user:
            return
        # don't respond to webhooks
        if message.author.discriminator == '0000':
            return
	# rhost api update channel
        #execstring="cemit({},escape({}#{} says, \"{}\"))".format(message.channel.name,message.author.display_name, message.author.discriminator, message.clean_content).encode('ascii', 'ignore').decode('ascii')
        execstring="cemit({},escape({}#{} says, \"{}\"))".format(message.channel.name,message.author.display_name, message.author.discriminator, emoji.demojize(message.clean_content))
        headers={"Exec":execstring, "Encode":"Yes"}
        print(execstring)
        requests.post("http://localhost:60601",headers=headers,auth=("#50","SomethingSecure"))
        #print(message.content)
        #curl -X POST --user "#50:SomethingSecure" -H "Exec:cemit(test,ApiBot says 'cemit hello from API')" --head http://localhost:60601

# don't mind me I'm just going to use 
# A RIDICULOUS AMOUNT OF MEMORY
# to be a relay
# because Discord is a stupid piece of shit
# that constantly demands websockets for anything useful
client = RhostRelayClient()
client.run('NjU3NzUxMDM2OTc0NDY1MDcy.Xf1weA.9MJzhdsLT_cM220pH40gaTJoABQ')

File: discord_receive.js
const Discord = require('discord.js')
const client = new Discord.Client()
const axios = require('axios')
const rhost = require('./rhost')

client.on('ready', () => {
	console.log(`Logged in as ${client.user.tag}!`)
})

client.on('message', message => {
	// don't respond to ourselves
	if(message.author == client.user)return
	// don't respond to webhooks
	if(message.author.discriminator == '0000')return
	const name = rhost.encodeString(`<${message.author.username}#${message.author.discriminator}>`)
	//const name = `<${message.member.displayName}#${message.author.discriminator}>`
	//Uncomment this, and comment the previous one to enable server nicknames instead.
	let post = rhost.encodeString(message.cleanContent)
	post = post.replace(/\r|\n|\r\n|\n\r/g,'%r%ch%cx%[cont%] ')
	if(message.attachments.array().length > 0){
		post = message.attachments.reduce((accumulate, attachment)=>`${accumulate}%r${attachment.url}`,`${post}%r- Attachments -`)
	}
	const execstring=`@cemitnp [udefault(me/chanhandler,${message.channel.name},${message.channel.name})]=%ch%cm[D]%cn ${name} ${post}`
	const headers={"Exec":execstring, "Encode":"Yes", "Parse":"ansiparse"}

	axios({
		method: 'post',
		url: 'http://0.0.0.0:60601',
		headers: headers,
		auth:{
			username: '#50',
			password: 'SomethingSecure'
		}
	}).catch((e)=>{console.error(e.message)})
})

client.login('DISCORD BOT TOKEN')
